src/log4net/Core/Level.cs

```diff
@@ -20,523 +20,468 @@
+  ///      <description>Meaning</description>
+  ///    </listheader>
   ///    <item>
-  ///      <description><see cref="Info"/>.</description>
+  ///      <term>Less than zero</term>
+  ///      <description><paramref name="l" /> is less than <paramref name="r" />.</description>
   ///    </item>
   ///    <item>
-  ///      <description><see cref="Debug"/>.</description>
+  ///      <term>Zero</term>
+  ///      <description><paramref name="l" /> is equal to <paramref name="r" />.</description>
   ///    </item>
   ///    <item>
-  ///      <description><see cref="All"/>.</description>
+  ///      <term>Greater than zero</term>
+  ///      <description><paramref name="l" /> is greater than <paramref name="r" />.</description>
   ///    </item>
   /// </list>
-  /// </remarks>
-  /// <author>Nicko Cadell</author>
-  /// <author>Gert Driesen</author>
-  [Serializable]
-  public class Level : IComparable, ISerializable
+  /// </returns>
+  public static int Compare(Level? l, Level? r)
   {
-    /// <summary>
-    /// Constructor
-    /// </summary>
-    /// <param name="level">Integer value for this level, higher values represent more severe levels.</param>
-    /// <param name="levelName">The string name of this level.</param>
-    /// <param name="displayName">The display name for this level. This may be localized or otherwise different from the name</param>
-    /// <remarks>
-    /// <para>
-    /// Initializes a new instance of the <see cref="Level" /> class with
-    /// the specified level name and value.
-    /// </para>
-    /// </remarks>
-    public Level(int level, string levelName, string displayName)
+    if (ReferenceEquals(l, r))
     {
-      Value = level;
-      Name = string.Intern(levelName.EnsureNotNull());
-      DisplayName = displayName.EnsureNotNull();
+      return 0;
     }
 
-    /// <summary>
-    /// Constructor
-    /// </summary>
-    /// <param name="level">Integer value for this level, higher values represent more severe levels.</param>
-    /// <param name="levelName">The string name of this level.</param>
-    /// <remarks>
-    /// <para>
-    /// Initializes a new instance of the <see cref="Level" /> class with
-    /// the specified level name and value.
-    /// </para>
-    /// </remarks>
-    public Level(int level, string levelName)
-      : this(level, levelName, levelName)
-    { }
-
-    /// <summary>
-    /// Serialization constructor
-    /// </summary>
-    /// <param name="info">The <see cref="SerializationInfo" /> that holds the serialized object data.</param>
-    /// <param name="context">The <see cref="StreamingContext" /> that contains contextual information about the source or destination.</param>
-    /// <remarks>
-    /// <para>
-    /// Initializes a new instance of the <see cref="Level" /> class 
-    /// with serialized data.
-    /// </para>
-    /// </remarks>
-    protected Level(SerializationInfo info, StreamingContext context)
+    if (l is null && r is null)
```

Comment:
unnecessary - if both are null, `ReferenceEquals` returns true.

Reply:
Thanks for the hint. I fixed it.